### PR TITLE
Introduce index_files function not restricted to JLSO

### PR DESCRIPTION
**Pull request by nickrobinson251**

### CODE REVIEW

File: test/indexing.jl [hunk 1]
```diff
@@ -19,6 +19,19 @@
...
     end
 
+    @testset "files not saved by Checkpoints.jl" begin
+        mktempdir(SystemPath) do path
+            Checkpoints.config("TestPkg.bar", path)
+            TestPkg.bar([1,2,3])
+            other_file = joinpath(path, "date=2021-01-01", "one")
+            mkpath(dirname(other_file))
+            write(other_file, 1)
+            @show index = index_files(path)
```
Comment (nickrobinson251) [inline test/indexing.jl:29]:
```suggestion
            index = index_files(path)
```

File: test/indexing.jl [hunk 2]
```diff
@@ -19,6 +19,19 @@
         end
     end
 
+    @testset "files not saved by Checkpoints.jl" begin
+        mktempdir(SystemPath) do path
+            Checkpoints.config("TestPkg.bar", path)
+            TestPkg.bar([1,2,3])
+            other_file = joinpath(path, "date=2021-01-01", "one")
```
Comment (oxinabox) [inline test/indexing.jl:26]:
```suggestion
            other_file = joinpath(path, "date=2021-01-01", "other_file.csv")
```

**Review by oxinabox — APPROVED** [review]
I guess this is basically just as much code as it would be to make `index_checkpoint_files` use `index_all_file`

File: test/indexing.jl (hunk 2 above)
Comment (nickrobinson251) [reply test/indexing.jl:26]: but it won't be a csv so that seems confusing (although of no importance i guess)

Comment (oxinabox) [reply test/indexing.jl:26]: OK. I figured a realistic extension would be good to have
`other_file.txt` ?

Comment (oxinabox) [reply test/indexing.jl:26]: maybe is good as is

Comment (nickrobinson251) [reply test/indexing.jl:26]:
```suggestion
            other_file = joinpath(path, "date=2021-01-01", "other_file.txt")
```

**Pull request merged**